### PR TITLE
home-assistant: unpin and fix build dependencies

### PR DESCRIPTION
**Pull request by tjni**
## Description of changes

These changes will be needed once we switch over to building Python packages with [`build`](https://pypa-build.readthedocs.io/en/latest/).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Thanks. I'm on mobile currently, so I'll take a more thorough look tonight.

File: pkgs/servers/home-assistant/default.nix [hunk 1]
```diff
@@ -368,6 +370,9 @@ in python.pkgs.buildPythonApplication rec {
       ) relaxedConstraints)}
       pyproject.toml
     substituteInPlace tests/test_config.py --replace '"/usr"' '"/build/media"'
+
+    sed -i 's/setuptools[~=]/setuptools>/' pyproject.toml
+    sed -i 's/wheel[~=]/wheel>/' pyproject.toml
```
[the remark this answers is not in the document]
Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix:375]: That breaks as it expects one dependency per line.

File: pkgs/servers/home-assistant/intents.nix [hunk 2]
```diff
@@ -32,13 +34,23 @@ buildPythonPackage rec {
...
   };
 
+  patches = [
+    # https://github.com/home-assistant/intents-package/pull/1
+    (fetchpatch {
+      name = "unpin-setuptools-dependency.patch";
+      url = "https://github.com/home-assistant/intents-package/commit/fc03c1ec1001ba3b812f2eea670d1c447cf2acee.patch";
+      hash = "sha256-kTu1+IwDrcdqelyK/vfhxw8MQBis5I1jag7YTytKQhs=";
+    })
+  ];
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/intents.nix:44]: Good luck getting anything merged after having mentioned you require this for nixpkgs.

**Comment by mweinelt** [comment]
Might as well go in as is. I'll have to clean up the whole dependency versioning story anyway.

**Comment by tjni** [comment]
Merged in https://github.com/NixOS/nixpkgs/pull/248866.

**Pull request closed without merging**